Feature: Allow devs to pass arguments to the node command via env variable

oigewan opened the pull request:
**Feature:**
While working on some code in the Node server in a previous PR, I found it very helpful to enable Node's [built-in inspector/debugger](https://nodejs.org/en/docs/inspector/).  Just in case you're not aware, here's a [good article](https://medium.com/@paul_irish/debugging-node-js-nightlies-with-chrome-devtools-7c4a1b95ae27#.pmqejrn8q) on what that does.  However, in order to enable this feature, I had to manually edit the Makefile to add the `--inspect` flag to the run target, then be sure to undo that change before committing.  It was a bit annoying.

I'd like to add a separate make target which does the same thing as `run`, but it also passes the `--inspect` flag to Node.  

For reference [Visual Studio Code](https://code.visualstudio.com/Docs/editor/debugging) supports this as a debug configuration, anyone can hit a URL shown on the command line in Chrome/Chromium to use the browser's development tools, and the standard debugger gets started on port 9229.

**Notes:**
I thought about conditionally adding the flag based on an environment variable, but I'm too new to the code base and I wanted to be safe by making this feature explicitly opt-in through a separate make target.

**Changes:**
- Add a new make target called `make run-inspect` which starts Node with the `--inspect` flag.
- Update docs to note the new target/possible command and its usage.

cc: @yoavf @dmsnell

dmsnell commented:
Thanks for the PR @oigewan! I love having docs on how to debug the node part of the build and the server-side rendering piece. There were a few things which gave me pause as I read through this:

Is there another way to accomplish passing additional args to the `make run` process without creating a new Make target for each other? Something like `make run --inspect` or `DEBUG=true make run` or anything along those lines…

The wording in the READMEs seems a wee bit informal. It also makes the assumption that "We all love the built-in development tools…" which could seem off-putting. This wouldn't be a PR blocker but I think that the wording could use some editing. Some things like "Now node allows…" are even a bit misleading since we've been able to debug node processes for a long time.

I could imagine this paragraph changing from dealing specifically with the server code to the more generic `node` debugging. For example, _Debugging the node process_, and mentioning that both the initial build process can connect to a remote debugger and also the server in the server-side render part.

Finally, please note that this is specific to `--inspect` when `--debug` and `--debug-brk` also exist. Kind of wraps up to my first hesitation: if we want to support those do we have to also create separate Make targets or is there a way we could get options passed in?

P.S. I realize that Make isn't really our friend when it comes to passing in arguments so I'm not sure if it's even practical for us to get it working the way I'm suggesting.

Thanks!

oigewan commented:
@dmsnell I totally agree.  My first instinct was to pass in something, but, as you said, Make isn't our friend when it comes to passing in args.  I chose `--inspect` over the other options since that allows the chrome inspector to do its things and starts up the classic debugger on a port.  Doesn't support `debug-brk`, but...

The environment variable method is probably the only decent way to pass in args.  I can play with that.

As for the docs, I admit that I'm not exactly the greatest technical writer in the world.  I'll give it another crack based on your feedback.

dmsnell commented:
> I admit that I'm not exactly the greatest technical writer in the world

no worries @oigewan - if you are comfortable updating the code but uncomfortable with the writing then you are more than welcome to ask for some help. I think oftentimes we might hesitate to jump in on someone else's work as we try to be courteous to each other, but I often do ask for help from someone else who is more capable than me for a given task. that being said, the docs are fine and it's mainly just some stylistic choices I think we could update that would improve them.

oigewan commented:
@dmsnell No problem on the docs.  Not saying I'm not comfortable, just saying it's probably a good thing that you're looking at what I write :smiley: 

I've updated and rebased with a more general solution.  This one doesn't have a separate make target and instead opts to just pass whatever's in an environment variable through to node.  I've also updated the docs I wrote.

oigewan commented:
ping @dmsnell 

This has been rebased.  Pinging so that the PR doesn't get lost in the shuffle.

nb commented:
The lint passes locally, it sounds like something with CircleCI, will look into more.

nb commented:
The same commit passes linting in #13883, @dmsnell, feel free to use the other branch to merge, or if there are more changes here by @oigewan, just sync the branch with the one in Calypso. Sorry for the convoluted process, we‘re working on fixing the linter :) 👨‍⚕️

nb commented:
Since the build is passing now, it doesn't seem like it’s the linter, but probably a CircleCI problem, so we’re back to this PR.

dmsnell commented:
Thanks again @oigewan! Sorry for the delay at the end. We're happy to have you as a contributor!
